docs/docs/concepts/how-it-works/spfx-project-upgrade.mdx

```diff
@@ -0,0 +1,227 @@
+
+Additionally, each finding has one or more **finding instances**.
+
+### Finding instance
+
+A finding instance describes the name of the file, and the location in the file where the finding's resolution should be applied.
+
+Typically, each finding has one finding instance, but there are some rules that are related to code changes that need to be applied multiple times, which can lead to a finding having multiple instances.
+
+### Report
+
+When the `spfx project upgrade` command ends its execution, it presents the user with a report. The report contains the list of changes that need to be applied to the project to upgrade it to the desired SPFx version (findings).
+
+The `spfx project upgrade` command offers several types of reports, each optimized for a different scenario:
+
+- **JSON** with structured output, recommended for automated processing
+- **plain-text** with basic human-readable output
+- **markdown** with rich human-readable output, recommended for most cases
+- **CodeTour** for interactive upgrade guidance using the VSCode CodeTour plugin, recommended if you want to learn more about the changes and easily navigate through your project
+
+Report types are supported through command output in the `spfx project upgrade` command.
+
+### Project model
+
+To speed up the upgrade process, the `spfx project upgrade` command starts the upgrade by collecting the relevant project files and building an object model out of them. That way, files need to be parsed only once and rules can be applied on in-memory objects that represent the project. This makes the upgrade process fast, even when it spans multiple versions with many different changes.
+
+Project model is stored in `src/m365/spfx/commands/project/project-model`.
+
+### Exit codes
+
+Shell executables (CLIs, scripts, etc.) communicate to the parent process whether their execution succeeded using exit codes. An exit code of 0 means success and a non-0 exit code means failure. The `spfx project upgrade` command uses exit codes to communicate different error states to the parent process using it. These error codes have been originally designed for use with the PnP Yeoman Generator.
+
+The exit codes of the `spfx project upgrade` command are defined in the `src/m365/spfx/commands/project/project-upgrade.ts` file.
+
+## How it works
+
+This section explains the process that the `spfx project upgrade` command follows to produce steps to upgrade the SPFx project to the specified SPFx version.
+
+![Diagram illustrating the process of the spfx project upgrade command](../../images/how-it-works/spfx-project-upgrade-process.png)
+
+### Detecting project version
+
+The `spfx project upgrade` command starts by checking if it's been executed in a folder belonging to an SPFx project.
+
+Once the command located the SPFx project's root folder, it will try to determine the project's version. It does this by using the following steps:
```

Comment:
```suggestion
Once the command locates the SPFx project's root folder, it will try to determine the project's version. It does this by using the following steps:
```